sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/implementation/PartitionProcessorImpl.java

```diff
@@ -207,8 +232,16 @@ public RuntimeException getResultException() {
         return this.resultException;
     }
 
-    private Mono<Void> dispatchChanges(FeedResponse<JsonNode> response) {
-        ChangeFeedObserverContext context = new ChangeFeedObserverContextImpl(this.settings.getPartitionKeyRangeId(), response, this.checkpointer);
+    private Mono<Void> dispatchChanges(
+        FeedResponse<JsonNode> response,
+        ChangeFeedState continuationState) {
+
+        ChangeFeedObserverContext context = new ChangeFeedObserverContextImpl(
+            // TODO fabianm - move observer to FeedRange for merge support
+            ((FeedRangePartitionKeyRangeImpl)this.settings.getStartState().getFeedRange()).getPartitionKeyRangeId(),
```

Comment:
there's some significant de-ref'ing here; might be worth to move this in a "try/catch" and add specific trace logging in case we fail then through the exception.

Reply:
Fixed (next iteration)